Core/Core/MobileCore.cs

```diff
@@ -196,14 +196,8 @@ private T GetInstance<T>(Type serviceClass, ServiceConfiguration serviceConfigur
             {
                 return (T)services[serviceClass];
             }
-
-            IServiceModule serviceModule = Activator.CreateInstance(serviceClass) as IServiceModule;
-            var serviceCfg = serviceConfiguration;
-            if (serviceCfg == null) serviceCfg = GetServiceConfiguration(serviceModule.Type);
-            if (serviceCfg == null && serviceModule.RequiresConfiguration) throw new ConfigurationNotFoundException($"{serviceModule.Type} not found on " + ConfigFileName);
-            serviceModule.Configure(this, serviceCfg);
-            services[serviceClass] = serviceModule;
-            return (T)serviceModule;
+            // There are no services registered for this interface.
+            throw new ServiceModuleInstanceNotFoundException(serviceClass.Name);
```

Comment:
@ziccardi Does this make sense to you? I was seeing the `Could not create instance of IAuthService` error before I registered the AuthService implementation. I think this error is more descriptive

Reply:
Yes, I would even add a message like: String.Format("No instance has been registered for interface {0}", serviceClass.Name)